libpod/pod_api.go

```diff
@@ -96,36 +96,42 @@ func (p *Pod) Stop(cleanup bool) (map[string]error, error) {
 	// dependencies. Should we bother with this?
 
 	// Stop to all containers
-	for _, ctr := range allCtrs {
-		ctr.lock.Lock()
 
-		if err := ctr.syncContainer(); err != nil {
-			ctr.lock.Unlock()
-			ctrErrors[ctr.ID()] = err
-			continue
-		}
-
-		// Ignore containers that are not running
-		if ctr.state.State != ContainerStateRunning {
-			ctr.lock.Unlock()
-			continue
-		}
+	size := len(allCtrs)
+	errs := make(chan bool, size)
+	for _, c := range allCtrs {
+		go func(ctr *Container) {
+			defer func() { errs <- true }()
```

Comment:
This seems like a wait group thing - https://golang.org/pkg/sync/#WaitGroup

Reply:
Did not know about that.  Thanks.